mesh/mesh-io-uart.c

```diff
@@ -321,6 +334,12 @@ static bool uart_io_init(struct mesh_io *mesh_io, void *opts,
 	mesh_io->pvt->ready_callback = cb;
 	mesh_io->pvt->user_data = user_data;
 
+	mesh_io->pvt->token_bucket = l_new(struct token_bucket, 1);
+	if (!mesh_io->pvt->token_bucket)
+		return false;
+
+	token_bucket_init(mesh_io->pvt->token_bucket, packets_per_second);
```

Comment:
Wrap allocation in `token_bucket_new` and return the pointer. This way the structure can be made private.